### PR TITLE
NAS-127110 / 24.10 / ix-table-columns-selector columns -> reset to default hides all columns/rows values & Select/Unselect all works with glitches

### PR DESCRIPTION
**Pull request by AlexKarpov98**
Testing: (see ticket & videos inside the ticket - for the bugs I've found)
See results 👇 

https://github.com/truenas/webui/assets/22980553/a8512a23-5d4c-4c5c-ad0d-88791e88517b

### CODE REVIEW

File: src/app/modules/ix-table2/components/ix-table-columns-selector/ix-table-columns-selector.component.ts [hunk 1]
```diff
@@ -63,8 +68,13 @@ export class IxTableColumnsSelectorComponent<T = unknown> implements OnChanges {
 
   private setInitialState(): void {
     this.columns = _.cloneDeep(this.defaultColumns);
-    this.hiddenColumns.clear();
-    this.hiddenColumns.select(...this.columns.filter((column) => column.hidden));
+    this.hiddenColumns.select(...this.columns);
+
+    this.defaultColumns
+      .filter((column) => !column.hidden)
+      .map((column) => this.defaultColumns.findIndex((column2) => column.title === column2.title))
```
Comment (undsoft) [inline src/app/modules/ix-table2/components/ix-table-columns-selector/ix-table-columns-selector.component.ts:75]: Can we rewrite this in some other way, this is a bit of hard to follow.

**Review by undsoft — CHANGES_REQUESTED** [review]
So, what is `unselect all` supposed to do? Show empty table or show first column? On `Replication Tasks` 3 random columns that are not on the list.

Also would be good to update tests so that they alert us in case of future problems.

File: src/app/modules/ix-table2/components/ix-table-columns-selector/ix-table-columns-selector.component.ts (hunk 1 above)
Comment (AlexKarpov98) [reply src/app/modules/ix-table2/components/ix-table-columns-selector/ix-table-columns-selector.component.ts:75]: > Can we rewrite this in some other way, this is a bit of hard to follow.

sure, but only `findIndex` works, that's weird

**Comment by AlexKarpov98** [comment]
> So, what is `unselect all` supposed to do? Show empty table or show first column? On `Replication Tasks` 3 random columns that are not on the list.
> 
> Also would be good to update tests so that they alert us in case of future problems.

I think `unselect all` should give us at least first column, because if we show nothing - that's not looking good and not reasonable I think.
For me - how it works now - it's `good` especially comparing to the previous glitchy version of it.

**Comment by AlexKarpov98** [comment]
@undsoft  - I think we can rewrite logic for `Select all` & `Unselect all` 
Currently we have `Unselect all` when some items are selected, I think we shall see `Select all` unless we selected all items - then we can show `Unselect all` 

What do you think?

**Review by undsoft — APPROVED** [review]
Nice, selector looks good.
